### DISCUSSION TITLE
Forecast Question - Expected Rain

### DISCUSSION THREAD

**Discussion by cweakland** [poblabs/weewx-belchertown#793] [Q&A]
I have my Belchertown skin up and functioning and I am using DarkSky for my forecast information. Is there a way to show the  foretasted rain amount? I have seen screenshots of it being done for snow fall, but my rain just displays a percentage of probability. 

Thanks, 
Chris

**Comment by norecords** [comment]
Hi Chris you have to modify the jQuery to display rain amount.
On my skin I show the amount only if the probability is more than 20%
For the 1 hour forecast, Replace
https://github.com/poblabs/weewx-belchertown/blob/026619d3230843d394edb33a657f32ffe6790de9/skins/Belchertown/js/belchertown.js.tmpl#L1341-L1345
By
```
                    } else if (forecast_row[i]["precip"] > 0) {
                        output_html += '<i class="wi wi-raindrop wi-rotate-45 rain-precip"></i>';
                        if (forecast_row[i]["precip"] > 20) {
                            output_html += '<span> ' + parseFloat(forecast_row[i]["precipMM"]).toFixed(0) + ' mm</span>';
                        } else {
                            output_html += '<span> ' + parseFloat(forecast_row[i]["precip"]).toFixed(0) + '%</span>';
                        }
                    } else {
                        output_html += '<i class="wi wi-raindrop wi-rotate-45 rain-no-precip"></i> <span>0%</span>';
                    } 
```
and For the 3/24 hour forecast, Replace
https://github.com/poblabs/weewx-belchertown/blob/026619d3230843d394edb33a657f32ffe6790de9/skins/Belchertown/js/belchertown.js.tmpl#L1387-L1391
By
```
                    } else if (forecast_row[i]["precip"] >= 20) {
                        output_html += '<i class="wi wi-raindrop wi-rotate-45 rain-precip"></i> <span>' + parseFloat(forecast_row[i]["precip"]).toFixed(0) + '%</span>';
                        output_html += '<span> | ' + parseFloat(forecast_row[i]["precipMM"]).toFixed(0) + ' mm</span>';
                    } else {
                        output_html += '<i class="wi wi-raindrop wi-rotate-45 rain-no-precip"></i> <span>0%</span>';
                    }
```
Note that is for Aeris forecast but should be the same from Darksky, check your forecast.json for `precipMM`.

**Comment by OE3MNS** [comment]
Hello,
I get NaN mm displayed in the prediction. Is there anything else that needs to be changed? I have precipMM in Forecast.json .

Thanks
